src/explore-education-statistics-common/src/modules/table-tool/utils/mapTableToJson.ts

```diff
@@ -39,25 +41,37 @@ export interface TableJson {
   tbody: TableCellJson[][];
 }
 
+interface Props {
+  tableHeadersConfig: TableHeadersConfig;
+  subjectMeta: FullTableMeta;
+  results: TableDataResult[];
+  query?: ReleaseTableDataQuery;
+}
+
 /**
  * Used to convert a table into a JSON representation of the HTML table
  * that can be used to render the table in the frontend.
- *
- * @param tableHeadersConfig
- * @param subjectMeta
- * @param results
- * @param query
- * @returns tableJson: TableJson, showMissingRowsOrColumnsWarning?: boolean
  */
-export default function mapTableToJson(
-  tableHeadersConfig: TableHeadersConfig,
-  subjectMeta: FullTableMeta,
-  results: TableDataResult[],
-  query?: ReleaseTableDataQuery,
-): {
+export default function mapTableToJson({
+  tableHeadersConfig: initialTableHeadersConfig,
+  subjectMeta,
+  results,
+  query,
+}: Props): {
   tableJson: TableJson;
-  showMissingRowsOrColumnsWarning?: boolean;
+  hasMissingRowsOrColumns?: boolean;
 } {
+  const tableHeadersConfig = {
+    columns: updateFilterGroup(initialTableHeadersConfig.columns),
+    columnGroups: initialTableHeadersConfig.columnGroups.map(filterGroup =>
+      updateFilterGroup(filterGroup),
+    ),
+    rows: updateFilterGroup(initialTableHeadersConfig.rows),
+    rowGroups: initialTableHeadersConfig.rowGroups.map(filterGroup =>
+      updateFilterGroup(filterGroup),
+    ),
+  };
```

Comment:
Do we need to re-map the table headers here? Whilst additional transforms things might be necessary sometimes, it'd be good to avoid doing more than we need to.

I've given the code a little look and it seems like we'd be able to do any additional work in `mapFullTableMeta` instead of the above implementation.

Reply:
Good call, it's now handled much more simply in `mapFullTableMeta`.